packages/ui/src/components/List/ListContext.tsx

```diff
@@ -209,6 +210,68 @@ export const ListProvider = ({
     ],
   )
 
+  const [lastCheckedIndex, setLastCheckedIndex] = useState<null | number>(null)
+
+  useEffect(() => {
+    const checkboxes = document.querySelectorAll<HTMLInputElement>(
+      '[name="list-select-checkbox"]',
+    )
```

Comment:
This won't work if you have multiple list / table in the same page. You need to find a more specific way to target it. Maybe the ref is the solution


https://github.com/user-attachments/assets/345d781c-6b91-4548-913f-07f4314ce37e